helpers/deproxy.py

```diff
@@ -660,7 +660,7 @@ def clear(self):
         self.request_buffer = ''
 
     def set_request(self, message_chain):
-        if message_chain:
+        if message_chain and message_chain.request:
```

Comment:
Ok, I see that the function is not safe, since `message_chain.request` could be `None` (which never happened though). But does the function does the same after the change? If `set_request()` is  called, then a `Client` should drop away current request and proceed  with the next one. Imagine, the `message_chain.request == None`, in this case previous request is removed and client begins to wait for unpaired responses. 
With this change nothing happens and on a next iteration client will repeat a request sent last time. This will hard to track.

So the main question here is intention. Is `message_chain.request == None` is allowed argument here? If yes, then fix only null pointer dereference couple lines below at `message_chain.request.msg`. If not, then raise an internal test suite error to allow us catch up and investigate the problem. But silent ignoring is not an option.